refactor(compiler-cli): improve error for non-exported non-standalone

dario-piotrowicz opened the pull request:
improve the error message for non-standalone components which are not
exported from their module, and that are also imported directly as if
they were standalone

this change simply adds the suggestion to the developer to import the
ngModule instead

resolves #46004

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## Issue

Issue Number: #46004



## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
 - I think/hope this resolves #46004, unless I understood the issue wrongly :sweat: 
 
This is the result:
![Screenshot at 2022-05-23 23-06-47](https://user-images.githubusercontent.com/61631103/169913262-5f790668-dc68-48c1-9360-d8a99fb33e7c.png)

dario-piotrowicz committed:
refactor(compiler-cli): improve error for non-exported non-standalone

improve the error message for non-standalone components which are not
exported from their module, and that are also imported directly as if
they were standalone

this change simply adds the suggestion to the developer to import the
ngModule instead

resolves #46004

dario-piotrowicz commented on packages/compiler-cli/test/ngtsc/standalone_spec.ts:
```diff
@@ -369,14 +369,12 @@ runInEachFileSystem(() => {
            expect(diags[0].code).toBe(ngErrorCode(ErrorCode.COMPONENT_IMPORT_NOT_STANDALONE));
            expect(getSourceCodeForDiagnostic(diags[0])).toEqual('TestDir');
 
-           // The diagnostic produced here should suggest that the directive be imported via its
-           // NgModule instead.
```
@alxhub is this simple change what you had in mind with this comment? :slightly_smiling_face: 

(or am I missing some nuance here? :thinking:)

AndrewKushnir reviewed (commented):
@dario-piotrowicz sorry for the delay, the change looks great! I've just added 1 quick comment and I can re-review and LGTM the change once the feedback is addressed. Thank you.

dario-piotrowicz commented:
Thanks @AndrewKushnir :heart:, no problem for the delay, thanks for having a look at it :blush: 

I've applied your suggestion :slightly_smiling_face::+1:

AndrewKushnir commented:
@dario-piotrowicz thanks for addressing the feedback. It looks like there are some tests that require an update due to the error message change. Please let me know once it's fixed and I can LGTM the PR and add it to the merge queue. Thank you.

dario-piotrowicz commented:
> @dario-piotrowicz thanks for addressing the feedback. It looks like there are some tests that require an update due to the error message change. Please let me know once it's fixed and I can LGTM the PR and add it to the merge queue. Thank you.

Ah yeah, sorry about that, I just noticed the failures :sweat: , I'll look into them and fix them tomorrow :bow:

AndrewKushnir commented:
This PR was merged into the repository by commit 3562eb107d9ae9b2e15729656e3fe1aafde3553e.
